### DISCUSSION TITLE
Opening a general discussion around the paid customer experience, general communication, breaking changes

### DISCUSSION THREAD

**Discussion by LucasZapico** [twentyhq/twenty#7081] [Unsolvable issues]
## Story

I'm currently paying for the hosted Twenty services and I woke up today to find my contact forms were not working. After some debugging I discovered the shape of the email object was changed via this PR https://github.com/twentyhq/twenty/pull/6885.

I wanted to callout this issue as Twenty is being sold as production ready on the open market for teams that may not be technical. I believe some versioning processes and CX update process guidelines is in order to ensure a customer doesn't suddenly have broken operations when the core API changes on a PR.

### Solutions Suggestions

Update the README and marketing copy on the site and convey that this product is in BETA and breaking changes may occur at any time and it is not recommended for nontechnical teams who are not following PRs.
Have a release schedule for the paid version so customers are informed of breaking changes via email if they are not watching PRs.

These two suggestions are not mutually and it may be best if both are implemented. Definitely not a good experience for a paying customer if the expectations are not set early. In my opinion, Twenty's marketing material makes is sound more production-ready then it is.

[Link to original issue](https://github.com/twentyhq/twenty/issues/7037)

Thanks for the acknowledgment already.

**Comment by FelixMalfait** [comment]
Hey @LucasZapico apologies again for this, I added a credit balance to your Stripe account so you shouldn't be charged for the next few months.

We've got another migration planned soon which is similar to EMAIL, the PHONE field type will become PHONES.
We'll adapt the process and give a notice for that one.

Twenty is marked as "alpha" on the README but you'd have to read through all the text. I think what we really want is not to hide between alpha/beta but just change our process to avoid introducing big breaking changes without notice. 
We've never emailed our client base because we don't want to bother people, so we need to figure out what's the best way to do this. I think the best would be to improve our data analysis tools and everytime a breaking change occurs, we should email only the users that we detect will be affected by this change...

**Reply by LucasZapico** [reply]
@FelixMalfait Thank you for the credit 😳. I really just want Twenty to do well. 

### Data Structure Changes 

Got it, makes sense HubSpot and Zoho have similar contact data structures. 

### README Communication 

The fact that Alpha is explicitly called out in the README is on me then.  I guess I just got excited by the prospect of not having to use HubSpot, Zoho or one of the dated poorly maintained open source self-hosted CRM options. 

I understand and really appreciate that you don't want to email people.... honestly there is a good chance I would miss the email anyway. 

### A Few More Solutions Suggestions 

I know it's a non-trivial labor cost but I wanted to callout [Vscodes](https://code.visualstudio.com/updates/v1_93) living and public change log. 

Something maybe worth considering is convey update in the app somehow. There are quite a few design patterns that could solve this, some more intrusive and others less.  This communication could also be tailored  by customer  given you have API usage by customer. Honestly some cool options with different dialog patterns.  I'm sure the Twenty team will come up with something that makes sense for where you guys are at. 

#### Discord Channel that has development 

This is then opt in for customers but is still easier than going through PRs... depending on how it's summarized. 

### Closing 

I know ideas are easy and implementation and labor management is hard. Thank you for taking the time to dialog, maybe some of these suggestions can develop into something of value.

**Reply by du8die** [reply]
FWIW - "We've never emailed our client base because we don't want to bother people" - bother me. If there are breaking changes, please bother me. Email changelogs (there are many products out there that can help). I can understand PR's and stuff - but I'd much rather read plain English explainations for what was changed.

**Reply by FelixMalfait** [reply]
@du8die thanks for the feedback! What I meant was not that we don't want to do it, but since we haven't sent any email, it felt a bit awkward to start with a first email that talks about something anecdotal that would impact only a small portion of users. But 100% aligned with you: we need to communicate more. This and stability are our top priorities next year. For long we've felt that adding more features should be the priority, now with Workflows coming out soon which will be a very powerful feature, it'll make sense to consolidate the existing scope and produce more content to explain how to use the platform, what's changing, etc.

We've been trying not to introduce breaking changes since this issue was raised. Did you notice any we've missed? FYI we also maintain a plain-english changelog here: https://twenty.com/releases

**Reply by LucasZapico** [reply]
@FelixMalfait I just wanted to pump this discussion a little bit as I have run into a few more breaking changes to the api since I first posted. I spent some time to build out a suite of playwright tests that hit the endpoints I use in Twenty and my services endpoints so I can catch the changes on each build. I'm hoping to free up more time and help Twenty more directly in this. 

Sorry the post doesn't have something more tangible and actionable.

**Reply by FelixMalfait** [reply]
@LucasZapico want to have a call? https://cal.com/ifelix. We already have a suite of Playwright tests but it's not great, maybe we can work on something together?

**Reply by LucasZapico** [reply]
Just scheduled a time with you March 5th. I look forward to finding a way to support the project.
